Integrate LIFI - #3j8nfb4

gitChimp88 opened the pull request.

gitChimp88 committed:
Code basic layout ready for LIFI swap integration

gitChimp88 committed:
Add select token page, connect LIFI sdk showing token list and balances

gitChimp88 committed:
Style swap back button and title, set from and to when clicked on

gitChimp88 committed:
Add ability to select chain and token, save in state and display

gitChimp88 committed:
Rename component to make less confusing

gitChimp88 committed:
Add search token filter, find routes and start display component

gitChimp88 committed:
Add info and styling to swap routes card

gitChimp88 committed:
Style and display details on swap route card

gitChimp88 committed:
Add selected route page, add temporary way to connect wallet

gitChimp88 committed:
Show swap status in realtime, listen to status and provide nav buttons

gitChimp88 committed:
Add navigation between swap details and routes

gitChimp88 committed:
Add active swap page with details of currently processing swaps

gitChimp88 committed:
Add option to clear all active swaps (old errors))

gitChimp88 committed:
Assign correct types to all components in the Dex feature

gitChimp88 committed:
Enable switch back and forth between tokens

gitChimp88 committed:
Add check before switching inputs

gitChimp88 committed:
Renamed DEX to Dex for deployment

gitChimp88 committed:
Change name of folder to fix git error

nunocgoncalves commented:
Task linked: [CU-3j8nfb4 Swap: Integrate LI.FI](https://app.clickup.com/t/3j8nfb4)

gitChimp88 committed:
Add gas and fund warning for the user

gitChimp88 committed:
Add ability to restart route and update switchChain

gitChimp88 committed:
Add powered by lifi

gitChimp88 committed:
Add logic so routes appear at correct time

make sure input doesn't show 0 on zero, allow floats and restrict to only numbers.

gitChimp88 committed:
Update switchChainHook and add monetisation to getRoutes

gitChimp88 committed:
Comment out monetisation until we have set it up with lifi

gitChimp88 committed:
Improve load time of token list, add infinite scroll

gitChimp88 committed:
Make re-usable card and modal component

Commit:
Merge branch 'main' into lifi-swap-copy

nunocgoncalves commented on .eslintrc:
```diff
@@ -32,6 +32,7 @@
     "react/jsx-key": "off",
     "react/no-unescaped-entities": "off",
     "linebreak-style": 0,
+    "no-async-promise-executor": "off",
```
Where are we using asynchronous functions as Promise executors and do we need to? If so, what are the drawbacks.

gitChimp88 replied:
Ah i'll turn this back on, I think I was just experimenting with something. But to answer the question executor functions shouldn't be async you can find the drawbacks here - https://eslint.org/docs/latest/rules/no-async-promise-executor

nunocgoncalves commented on src/components/Account/connectors/networks.ts:
```diff
@@ -0,0 +1,110 @@
+import type { AddEthereumChainParameter } from "@web3-react/types";
```
We're not using this anywhere, right?

gitChimp88 replied:
Yes it's used in TemporaryConnectAccount.tsx line 91 in function getAddChainParameters, also just a side note don't worry too much about this code because we will delete it once we have the authentication up and  running

nunocgoncalves commented on src/components/Dex-lifi/SwapRoutes/utils.ts:
```diff
@@ -0,0 +1,46 @@
+import type { Route, Token } from "@lifi/sdk";
+import Big from "big.js";
+
+export const getGasCostsBreakdown = (route: Route) => {
```
Do we get the gas estimations from the LI.FI API when we are requesting a route? From what I read in the code, it seems like that's the case but I couldn't find anything in the API reference.

gitChimp88 replied:
Yeah the gas cost is returned from the api in the useSwapRoutes hook, line 120. You can actually ctrl click on life.getRoutes then ctrl click on RoutesResponse then you will see the Route interface documenting what's contained within the route.

nunocgoncalves commented on src/providers/LIFIProvider.tsx:
```diff
@@ -0,0 +1,26 @@
+    const config = {};
+    if (!lifi) {
+      lifi = new LIFI({
+        disableVersionCheck: true,
```
When enabled, does this property make you register the SDK/API version?

gitChimp88 replied:
I'm not entirely sure on this will have to check with LIFI, the config parameter is completely optional I just left it here in case we wanted to edit this part and customise it later.

gitChimp88 committed:
Add various UI fixes

gitChimp88 committed:
Merge branch 'lifi-swap-copy' of https://github.com/Pizza-Wallet-Development-team/pizza-wallet into lifi-swap-copy

gitChimp88 committed:
Exclude unsupported chains from swap

gitChimp88 committed:
Fix UI bugs for swap

nunocgoncalves commented:
Code in ``transfer-assets`` branch
